### PR TITLE
Add new SocialPostBadge component

### PR DESCRIPTION
**Pull request by aprentout**
### What does this PR do?

Related to: [#2700](https://github.com/upfluence/backlog/issues/2700)

### What are the observable changes?
New SocialMediaBadge component based on existing one from hyper stream

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: translations/en-us.yaml [hunk 1]
```diff
@@ -55,3 +55,14 @@ oss-components:
...
+    facebook_status: Facebook Status
+    tweet: Tweet
+    pin: Pin
+    instagram_media: Instagram Media
+    tiktok_video: Tiktok Video
+    youtube_video: Youtube Video
+    article: Article
+    story: Story
+    twitch_stream: Twitch Stream
+    any: Any
```
Comment (phndiaye) [inline translations/en-us.yaml:68]: you can remove this one as it has a meaning in Hyperstream but not really here :)

File: tests/integration/components/o-s-s/social-media-badge-test.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import { click, render } from '@ember/test-helpers';
+import hbs from 'htmlbars-inline-precompile';
+import setupOnerror from '@ember/test-helpers/setup-onerror';
```
Comment (phndiaye) [inline tests/integration/components/o-s-s/social-media-badge-test.ts:5]: `setupOnerror` can be imported in line 3 also, btw :)

File: addon/components/o-s-s/social-media-badge.hbs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<div class={{this.computedClasses}} role="button"
+     {{enable-tooltip title=(t (concat "oss-components.post_types." @postType))}}
+     {{on "click" this.onPostTypeClick}}
+     ...attributes>
+    <div class="icon-container">
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.hbs:5]: I think your IDE config is wrong, this looks like '4 chars' for spaces but we use 2 chars.
Also the alignment of the div attributes seems weird

File: addon/components/o-s-s/social-media-badge.stories.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import hbs from 'htmlbars-inline-precompile';
+
+const PostTypes = [
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.stories.js:3]: These are not Post types :)

File: addon/components/o-s-s/social-media-badge.stories.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  'twitch_stream'
+];
+
+export default {
+  title: 'Components/OSS::SocialMediaBadge',
+  component: 'socialMediaBadge',
+  argTypes: {
+    postType: {
+      type: { required: true },
+      description: 'Type of the post used to load icon & style',
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.stories.js:21]: same as above, "post" is not the right word

File: addon/components/o-s-s/social-media-badge.stories.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        type: {
+          summary: 'onToggle(postType: String): void'
+        }
+      }
+    }
+  },
+  parameters: {
+    docs: {
+      description: {
+        component: 'Displays a media icon in a round badge.'
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.stories.js:65]: "a social media icon" ; but they are not really icons, more "skins"

File: addon/components/o-s-s/social-media-badge.stories.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  postType: 'pin',
+  plain: false,
+  selected: false,
+  onToggle: undefined
+};
+
+const Template = (args) => ({
+  template: hbs`
+    <OSS::SocialMediaBadge @postType={{this.postType}} @onToggle={{this.onToggle}} @selected={{this.selected}}
+                @plain={{this.plain}} />
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.stories.js:81]: alignment

File: addon/components/o-s-s/social-media-badge.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+
+import { assert } from '@ember/debug';
+
+export enum PostType {
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.ts:6]: Again, not PostType

What about a SocialMedia type instead ?

File: addon/components/o-s-s/social-media-badge.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    icon: 'fab fa-youtube'
+  },
+  {
+    key: PostType[PostType.twitch_stream],
+    icon: 'fab fa-twitch'
+  }
+];
+
+interface OSSSocialMediaBadgeArgs {
+  postType: PostType;
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.ts:60]: `skin` maybe ?
or socialMedia ? 🤷

Comment (Miexil) [reply addon/components/o-s-s/social-media-badge.ts:60]: or even just plain "type" would be better

File: app/styles/molecules/social-media-badge.less [hunk 10]
```diff
@@ -0,0 +1,171 @@
+.generate-social-media-badge(@social-media) {
```
Comment (Miexil) [inline app/styles/molecules/social-media-badge.less:1]: :/ 
We are mostly trying to get rid of using less, and although we still use it sometimes, I feel here is not a justified usage.
You already define css classes for the social media types (for example line 85/88) I think all of the common attributes should be in a common class that is always defined.

Rename `.generate-social-media-badge` to `social-media-badge`, give it default values for `background-color`, `border-color` and the .iconcontainer's `background`.
Then overwrite these three values in your social-media-types definition, something like : 
```css
  &--twitter {
    background-color: #color for twitter
    border-color: #color for twitter

    .icon-container {
        background: #background for twitter
    }
  }
```

Your computedClasses getter should already be working with this solution.

File: app/styles/molecules/social-media-badge.less [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+      -webkit-background-clip: unset;
+      -webkit-text-fill-color: unset;
+    }
+  }
+}
+
+.oss-social-media-badge {
+  transition: transform 0.25s ease-in-out;
+
+  &--tweet {
```
Comment (Miexil) [inline app/styles/molecules/social-media-badge.less:88]: twitter

File: app/styles/molecules/social-media-badge.less [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+      .generate-plain-social-media-badge('instagram');
+      background: var(--color-instagram-accent);
+
+      .icon-container  {
+        background: transparent;
+      }
+    }
+  }
+
+  &--story {
```
Comment (Miexil) [inline app/styles/molecules/social-media-badge.less:109]: `&__instagram_story` ? I think it's instagram, but not sure

File: app/styles/molecules/social-media-badge.less [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+  }
+
+  &--twitch_stream {
+    .generate-social-media-badge('twitch');
+    &.oss-social-media-badge--plain {
+      .generate-plain-social-media-badge('twitch');
+    }
+  }
+
+  &--pin {
```
Comment (Miexil) [inline app/styles/molecules/social-media-badge.less:143]: &--pinterest

File: tests/integration/components/o-s-s/social-media-badge-test.ts [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+          err.message,
+          'Assertion Failed: [component][OSS::SocialMediaBadge] The @postType parameter should be a value of PostType'
+        );
+      });
+      await render(
+        hbs`<OSS::SocialMediaBadge @postType={{this.postType}} @plain={{this.plain}} @selected={{this.selected}} @onToggle={{this.onToggle}}/>`
+      );
+      await settled();
+    });
+  });
```
Comment (Miexil) [inline tests/integration/components/o-s-s/social-media-badge-test.ts:58]: Move to the end of the file

File: translations/en-us.yaml (hunk 1 above)
Comment (Miexil) [inline translations/en-us.yaml:68]: missing line break
( I think your linter is misconfigured )

File: translations/en-us.yaml [hunk 15]
```diff
@@ -55,3 +55,14 @@ oss-components:
...
+  post_types:
+    facebook_status: Facebook Status
+    tweet: Tweet
+    pin: Pin
+    instagram_media: Instagram Media
+    tiktok_video: Tiktok Video
+    youtube_video: Youtube Video
+    article: Article
+    story: Story
+    twitch_stream: Twitch Stream
```
Comment (Miexil) [inline translations/en-us.yaml:67]: Don't know what the others think (@JulienVannier66 @phndiaye) but maybe we should let the parent assign or not a tooltip.
We might have a case where a SocialMediaBadge is not yet selected and the products wants us to put "Click on this badge to filter by Instagram posts".

File: addon/components/o-s-s/social-media-badge.stories.js [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+  'article',
+  'facebook_status',
+  'instagram_media',
+  'tiktok_video',
+  'story',
+  'tweet',
+  'pin',
+  'youtube_video',
+  'twitch_stream'
+];
```
Comment (Miexil) [inline addon/components/o-s-s/social-media-badge.stories.js:13]: Look at how we do things for typing in OSS::Button.

I suggest we use something like :
```typescript

export type SocialMediaType = 'article' | 'facebook_status' | 'instagram_media';
type skinDefinitionType = { [key in SocialMediaType]: string };
export const skinMatching: skinDefinitionType = {
  article: 'fab fa-wordpress',
  facebook_status: 'fab fa-facebook',
  instagram_media: ''
};
```

File: addon/components/o-s-s/social-media-badge.ts [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+    if (this.args.selected) {
+      classes.push('oss-social-media-badge--selected');
+    }
+
+    return classes.join(' ');
+  }
+
+  get iconDefinition(): string {
+    return mediaMatching.find((el: MatchingElement) => el.key === this.args.postType.toString())?.icon || '';
```
[the remark this answers is not in the document]
Comment (aprentout) [reply addon/components/o-s-s/social-media-badge.ts]: Change approach in 7efe4bdae28b942efbe5199b58241d1fe8a7b321

File: translations/en-us.yaml (hunk 1 above)
Comment (aprentout) [reply translations/en-us.yaml]: Removed in 7efe4bdae28b942efbe5199b58241d1fe8a7b321

File: tests/integration/components/o-s-s/social-media-badge-test.ts (hunk 2 above)
Comment (aprentout) [reply tests/integration/components/o-s-s/social-media-badge-test.ts]: Fix imports in ad4f3c0940b3fe3b00c6f3a59e1a2d4ebe7663e7

File: addon/components/o-s-s/social-media-badge.hbs (hunk 3 above)
Comment (JulienVannier66) [reply addon/components/o-s-s/social-media-badge.hbs:5]: Don't forget to add an empty line at the end of the file too

File: addon/components/o-s-s/social-media-badge.ts [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+export default class OSSSocialMediaBadge extends Component<OSSSocialMediaBadgeArgs> {
+  constructor(owner: unknown, args: OSSSocialMediaBadgeArgs) {
+    super(owner, args);
+
+    assert(
+      '[component][OSS::SocialMediaBadge] The @mediaType parameter of type SocialMediaType is mandatory',
+      this.args.mediaType
+    );
```
Comment (JulienVannier66) [inline addon/components/o-s-s/social-media-badge.ts:46]: It is really check the type here?

File: tests/integration/components/o-s-s/social-media-badge-test.ts [hunk 19]
```diff
@@ -0,0 +1,111 @@
...
+      setupOnerror((err: any) => {
+        assert.equal(
+          err.message,
+          'Assertion Failed: [component][OSS::SocialMediaBadge] The @mediaType parameter of type SocialMediaType is mandatory'
+        );
+      });
+      await render(
+        hbs`<OSS::SocialMediaBadge @plain={{this.plain}} @selected={{this.selected}} @onToggle={{this.onToggle}}/>`
+      );
+      await settled();
```
Comment (JulienVannier66) [inline tests/integration/components/o-s-s/social-media-badge-test.ts:93]: Really need that, same for the next test?

Comment (aprentout) [reply tests/integration/components/o-s-s/social-media-badge-test.ts]: removed

File: app/styles/atoms/social-post-badge.less [hunk 20]
```diff
@@ -0,0 +1,166 @@
+.oss-social-post-badge {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+
+  background-color: var(--color-white);
+  border: 1px solid;
```
Comment (JulienVannier66) [inline app/styles/atoms/social-post-badge.less:7]: Do you really need this border?

**Review by JulienVannier66 — APPROVED** [review]
Otherwise LGTM

File: translations/en-us.yaml [hunk 21]
```diff
@@ -55,3 +55,14 @@ oss-components:
...
+    media_types:
+      facebook_status: Facebook Status
+      tweet: Tweet
+      pin: Pin
+      instagram_media: Instagram Media
+      tiktok_video: Tiktok Video
+      youtube_video: Youtube Video
+      article: Article
+      story: Story
+      twitch_stream: Twitch Stream
```
Comment (Miexil) [inline translations/en-us.yaml:68]: This is unused :)

File: app/styles/atoms/social-post-badge.less (hunk 20 above)
Comment (aprentout) [reply app/styles/atoms/social-post-badge.less]: Removed :)

**Review by phndiaye — APPROVED** [review]

**Pull request merged**